File: Domain/Reservation.cs
using System;

namespace Domain
{
    public class Reservation
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }
        public string ReservationType { get; set; }
        public int status { get; set; }
        public AppUser user { get; set; }
    }
}

File: client-app/src/app/models/Reservation.ts
import { User } from "./user";


export interface Reservation {
    id: string;
    title: string;
    date: string;
    description: string;
    reservationType: string;
    status: number;
}

export interface ReservationUser {
    id: string;
    title: string;
    date: string;
    description: string;
    reservationType: string;
    status: number;
    user: User;
}